File: algorithms/searching/linear-search/LinearSearch.cpp
#include <vector>
using namespace std;

class Searching {
public:
  static int linearSearch(vector<int> &arr, int target) {
    for (int i = 0; i < arr.size(); i++) {
      if (arr[i] == target) {
        return i;
      }
    }

    return -1;
  }
};

File: algorithms/searching/linear-search/LinearSearch.dart
class Searching {
  static int linearSearch(List<int> array, int target) {
    int n = array.length;

    for (int i = 0; i < n; i++) {
      if (array[i] == target) {
        return i;
      }
    }

    return -1;
  }
}
